railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -141,6 +141,7 @@ def test
     end
 
     def tmp
+      empty_directory_with_keep_file "tmp"
```

Comment:
Or maybe we can leave this line as is and assume everywhere that we do have the tmp directory.